### PR TITLE
ndslice.selection: improve coverage

### PR DESCRIPTION
**Pull request by wilzbach**
let's improve mir's coverage!

### CODE REVIEW

File: source/mir/ndslice/selection.d [hunk 1]
```diff
@@ -1594,7 +1653,7 @@ template IndexSlice(size_t N)
     {
         private size_t[N-1] _lengths;
 
-        auto save() @property const
+        auto save() @property
```
Comment (wilzbach) [inline source/mir/ndslice/selection.d]: this is actually a bug. with `const` it isn't a forward range.

Comment (9il) [inline source/mir/ndslice/selection.d]: Why? Please add a test for IndexSlice for random access range.

File: source/mir/ndslice/selection.d [hunk 2]
```diff
@@ -325,6 +330,26 @@ Slice!(1, Range) diagonal(size_t N, Range)(auto ref Slice!(N, Range) slice)
...
+    //  -------
+    // | 0 1 |
+    // | 2 3 |
+    // | 4 5 |
+    //  -------
+    //->
+    // | 0 3 |
+
+    assert(6.iota
+        .sliced(3, 2)
```
Comment (9il) [inline source/mir/ndslice/selection.d]: `iotaSlice(3, 2)`

File: source/mir/ndslice/selection.d [hunk 3]
```diff
@@ -865,6 +890,20 @@ pure unittest
           [[[[ 2], [ 1], [0]]]]]]);
 }
 
+@safe pure unittest
+{
+    import mir.ndslice.slice;
+    import std.range: iota;
+    import std.exception: assertThrown;
+
+    auto e = 1.iota.sliced(1);
```
Comment (9il) [inline source/mir/ndslice/selection.d]: `iotaSlice(1)`

File: source/mir/ndslice/selection.d [hunk 4]
```diff
@@ -1201,6 +1240,11 @@ pure nothrow unittest
     elems.popBackExactly(2);
     assert(elems.back == 9);
     assert(elems.length == 8);
+
+    assert(elems.front == 2);
```
Comment (9il) [inline source/mir/ndslice/selection.d]: Please move this outside from documented unittest

File: source/mir/ndslice/selection.d (hunk 1 above)
Comment (wilzbach) [inline source/mir/ndslice/selection.d]: See line 1641 - it's transitive, so even `byElement` won't be possible otherwise.

Comment (9il) [inline source/mir/ndslice/selection.d]: have you tested it?

Comment (9il) [inline source/mir/ndslice/selection.d]: Agreed, my test fails)

Comment (9il) [inline source/mir/ndslice/selection.d]: please replace it with `IndexMap save() @property const`

File: source/mir/ndslice/selection.d (hunk 2 above)
Comment (wilzbach) [inline source/mir/ndslice/selection.d]: there are still a couple of tests using `iota.sliced` (e.g. the one above)

Comment (9il) [inline source/mir/ndslice/selection.d]: I have not fixed them in mir. Only in Phobos

Comment (9il) [inline source/mir/ndslice/selection.d]: I will update mir after Phobos PR will be merged

**Comment by wilzbach** [comment]
comments addressed - thanks for your input and feedback :)

**Comment by 9il** [comment]
Thank you for this PRs)

File: source/mir/ndslice/selection.d [hunk 5]
```diff
@@ -1527,6 +1577,9 @@ pure nothrow unittest
     import std.range: popFrontN;
     elems.popFrontN(3);
     assert(elems.front == 5);
+    import std.range: dropOne;
```
Comment (9il) [inline source/mir/ndslice/selection.d]: please use additional test

**Comment by 9il** [comment]
this should be rebased

**Comment by wilzbach** [comment]
rebased.

**Pull request merged**